drv/stm32h7-qspi/src/lib.rs

```diff
@@ -258,6 +282,18 @@ impl Qspi {
         // perform transfers.
         let mut out = out;
         while !out.is_empty() {
+            if self.reg.sr.read().tof().bit_is_set() {
```

Comment:
How about reading `self.reg.sr` just once, then checking `tof`, `tef`, and `flevel` (below) on that value?